### PR TITLE
Add defer idempotent messages

### PR DESCRIPTION
**Pull request by VDBBjorn**
Defered messages will be considered idempotent

### CODE REVIEW

**Comment by VDBBjorn** [comment]
Would this be considered a breaking change? Or is the updated version 1.0.1 -> 1.1.0 sufficient?

File: Rebus.Idempotency.MySql.Tests/Rebus.Idempotency.MySql.Tests.csproj [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
     <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.0.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.9.0" />
     <PackageReference Include="xunit" Version="2.2.0" />
     <PackageReference Include="xunit.runner.visualstudio" Version="2.2.0" />
-    <PackageReference Include="Rebus" Version="4.2.*" />
+    <PackageReference Include="Rebus" Version="5.4.*" />
```
Comment (dietermijle) [inline Rebus.Idempotency.MySql.Tests/Rebus.Idempotency.MySql.Tests.csproj]: Can't we immediately update to Rebus 6?

**Review by dietermijle — COMMENTED** [review]
My only suggestion would be to update to the latest version of Rebus.
Rest looks fine + some nice refactoring!

**Review by mario-d-s — APPROVED** [review]

File: Rebus.Idempotency.MySql.Tests/Rebus.Idempotency.MySql.Tests.csproj (hunk 1 above)
Comment (mario-d-s) [reply Rebus.Idempotency.MySql.Tests/Rebus.Idempotency.MySql.Tests.csproj]: @dietermijle agree
@VDBBjorn please keep in mind that libraries that depend on other libraries should ideally specify a version range, such as: `[4.0.0, 7.0.0)`

**Comment by mario-d-s** [comment]
> Would this be considered a breaking change? Or is the updated version 1.0.1 -> 1.1.0 sufficient?

Minor version seems enough for me.

**Review by almsyx — APPROVED** [review]

File: Rebus.Idempotency/Rebus.Idempotency.csproj [hunk 2]
```diff
@@ -24,11 +24,11 @@
   <PropertyGroup Condition=" '$(TargetFramework)' == 'net45' ">
     <DefineConstants>NET451</DefineConstants>
   </PropertyGroup>
-  <PropertyGroup Condition=" '$(TargetFramework)' == 'netstandard1.6' ">
+  <PropertyGroup Condition=" '$(TargetFramework)' == 'netstandard2.0' ">
     <DefineConstants>NETSTANDARD1_6</DefineConstants>
   </PropertyGroup>
   <ItemGroup>
-        <PackageReference Include="Rebus" Version="4.2.*" />
+        <PackageReference Include="Rebus" Version="5.4.*" />
```
Comment (mario-d-s) [inline Rebus.Idempotency/Rebus.Idempotency.csproj]: @VDBBjorn as mentioned please update this to a version range (covering the major versions we use in projects now).

**Review by mario-d-s — CHANGES_REQUESTED** [review]

**Comment by almsyx** [comment]
> > Would this be considered a breaking change? Or is the updated version 1.0.1 -> 1.1.0 sufficient?
> 
> Minor version seems enough for me.

As the package has updated the .NET standard version ( 1.6 -> 2.0), it is a breaking change, in my opinion.

**Comment by mario-d-s** [comment]
> > > Would this be considered a breaking change? Or is the updated version 1.0.1 -> 1.1.0 sufficient?
> > 
> > 
> > Minor version seems enough for me.
> 
> As the package has updated the .NET standard version ( 1.6 -> 2.0), it is a breaking change, in my opinion.

Good point, so yes we should update it!

File: Rebus.Idempotency/Rebus.Idempotency.csproj (hunk 2 above)
Comment (VDBBjorn) [reply Rebus.Idempotency/Rebus.Idempotency.csproj]: I've updated this to [5.4.0 - 7.0.0), as 5.4 is the version where Rebus introduced the defer headers.

File: Rebus.Idempotency.MySql.Tests/Rebus.Idempotency.MySql.Tests.csproj (hunk 1 above)
Comment (VDBBjorn) [reply Rebus.Idempotency.MySql.Tests/Rebus.Idempotency.MySql.Tests.csproj]: As Rebus introduced the defer headers only in version 5.4, I've updated the range to [5.4.0,7.0.0)

**Review by mario-d-s — APPROVED** [review]

**Pull request merged**